doc/designs/expiring-password-notification.md

```diff
@@ -0,0 +1,217 @@
+# Expiring Password Notifications
+
+**DESIGN STAGE**
+
+## Overview
+
+A method to warn users via email that their IPA account password is about to expire.
+Ticket [link](https://pagure.io/freeipa/issue/3687).
+
+## User Stories
+
+[0] As an IPA user, I want to be notified by email and through the WebUI when my password is near its expiry date so that I change my password before it expires.
+
+Outcome: Users whose passwords are expiring receive an email.
+NB: The WebUI warning already exists. Document (ipapwdexpadvnotify) usage.
+
+[1] As an IPA administrator, I want to be able to provide a template for the above emails so that they conform to company policy.
+
+Outcome: IPA administrators can edit an email template that is used for notifications.
+
+[2] As an IPA administrator, I want to be able to set a list of days before password expiry date on which these notifications are sent so that users are warned well in advance and often enough to change their passwords so that our IT does not get asked about it.
+
+Outcome: Users receive notifications on an IPA-admin-specified list of days before their password expires.
+Supersedes [1].
+
+[3] As an IPA administrator, I want to be able to set smtp client parameters so that the notification emails are sent properly.
+
+Outcome: Emails are sent according to the SMTP parameters set in the configuration file.
+
+[4] As an IPA administrator, I do not want to notify users whose accounts are disabled because it makes no sense to.
+
+Outcome: If an account is disabled no emails are sent when the password is near expiry.
```

Comment:
Worth specifically mentioning nsAccountLock or is that too much in the weeds?

Reply:
Done, thanks!